From f673fd36cdc17dac89f6504da48c51759451ff27 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 19 Aug 2022 15:20:53 +0300
Subject: [PATCH 085/151] drivers: net: lan865x: update reset gpio setting

- move to of_get_named_gpio() so standard gpio definition can be parsed
  from device tree

- make it optional

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 .../net/ethernet/microchip/lan865x/lan865x.c  | 24 +++++++++++--------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/drivers/net/ethernet/microchip/lan865x/lan865x.c b/drivers/net/ethernet/microchip/lan865x/lan865x.c
index e51f1e0d7e12..109612c292e4 100644
--- a/drivers/net/ethernet/microchip/lan865x/lan865x.c
+++ b/drivers/net/ethernet/microchip/lan865x/lan865x.c
@@ -19,6 +19,7 @@
 #include <linux/gpio.h>
 #include <linux/kthread.h>
 #include <linux/of.h>
+#include <linux/of_gpio.h>
 #include <linux/acpi.h>
 #include <linux/timer.h>
 #include <linux/jiffies.h>
@@ -79,7 +80,7 @@ struct lan865x_priv {
 	bool tc6NeedService;
 	bool tc6IntActive;
 	uint32_t read_value;
-	uint16_t reset_gpio;
+	int reset_gpio;
 	bool enable_data;
 	struct completion tc6_completion;
 	u8 plca_enable;
@@ -345,13 +346,15 @@ void TC6Stub_InitializeSpi(uint8_t idx, void *pGlobalTag)
 {
 	struct lan865x_priv *priv = pGlobalTag;
 
-	gpio_request(priv->reset_gpio, "reset_gpio");
-	gpio_direction_output(priv->reset_gpio, 1);
-	gpio_set_value(priv->reset_gpio, 0);
-	mdelay(50);
-	gpio_set_value(priv->reset_gpio, 1);
-	mdelay(50);
-	gpio_free(priv->reset_gpio);
+	if (priv->reset_gpio >= 0) {
+		gpio_request(priv->reset_gpio, "reset_gpio");
+		gpio_direction_output(priv->reset_gpio, 1);
+		gpio_set_value(priv->reset_gpio, 0);
+		mdelay(50);
+		gpio_set_value(priv->reset_gpio, 1);
+		mdelay(50);
+		gpio_free(priv->reset_gpio);
+	}
 }
 
 static bool lan865x_init(struct lan865x_priv *priv)
@@ -758,11 +761,12 @@ static int lan865x_get_of_data(struct lan865x_priv *priv)
 	if (!spi->dev.of_node)
 		return -EINVAL;
 
-	ret = of_property_read_u16(spi->dev.of_node, "reset-pin", &priv->reset_gpio);
-	if (ret < 0) {
+	ret = of_get_named_gpio(spi->dev.of_node, "reset-pin", 0);
+	if (ret < 0 && ret != -ENOENT) {
 		dev_err(&spi->dev, "reset-pin property is not found in device tree\n");
 		return ret;
 	}
+	priv->reset_gpio = ret;
 
 	ret = of_property_read_u8(spi->dev.of_node, "plca-enable", &priv->plca_enable);
 	if (ret < 0) {
-- 
2.17.1

